### PR TITLE
fix: decryptMLS crash if not apiV5

### PR DESCRIPTION
**Pull request by netbe**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

If somehow the user did not select apiV5 on a MLS build it will crash when receiving MLS messages.

Lower it to just ignore the event.

Notes:

Corrected the cartfile to show same version as Cartfile.resolved

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+MLS.swift [hunk 1]
```diff
@@ -27,8 +27,8 @@ extension EventDecoder {
         Logging.mls.info("decrypting mls message")
 
         guard let decryptionService = context.mlsDecryptionService else {
-            WireLogger.mls.critical("failed to decrypt mls message: mlsDecyptionService is missing")
-            fatalError("failed to decrypt mls message: mlsService is missing")
```
Comment (theswiftway) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+MLS.swift:31]: IMHO: we should replace all the fatalError to preconditionFailure. 

[details here](https://lapcatsoftware.com/articles/fatalError.html)

**Review by theswiftway — APPROVED** [review]
LGTM

**Review by johnxnguyen — COMMENTED** [review]
How do you repeat this crash? You somehow have to be part of MLS groups, yet not have API v5 selected?

**Comment by netbe** [comment]
> How do you repeat this crash? You somehow have to be part of MLS groups, yet not have API v5 selected?

Exactly if you stay on api v4 or if you don’t set apiv5 before the first login. Setup of core crypto don’t wait for api resolver to be finished or the network stack to be ready

**Comment by johnxnguyen** [comment]
> Exactly if you stay on api v4 or if you don’t set apiv5 before the first login. Setup of core crypto don’t wait for api resolver to be finished or the network stack to be ready

@netbe so in practice this is only an issue for developers and not in a real life situation?

**Comment by netbe** [comment]
> > Exactly if you stay on api v4 or if you don’t set apiv5 before the first login. Setup of core crypto don’t wait for api resolver to be finished or the network stack to be ready
> 
> @netbe so in practice this is only an issue for developers and not in a real life situation?

@johnxnguyen I see your point, I would say Developers + testers, maybe there's a better fix then to avoid this case

**Pull request closed without merging**